client/state/comments/models/comment.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * Internal dependencies
+ */
+import { Author } from './comment-author'; // eslint-disable-line no-unused-vars
+import { CommentInfo } from './comment-info'; // eslint-disable-line no-unused-vars
+
+export function Comment( constructor ) {
+	Object.defineProperty( this, 'constructor', {
+		value: constructor || this
+	} );
+}
+
+export function LoadingComment( siteId, commentId ) {
+	const comment = new Comment( LoadingComment );
+
+	comment.siteId = siteId;
+	comment.commentId = commentId;
+
+	return comment;
+}
+
+/**
+ * Create a new loaded comment value
+ *
+ * @param {Number} siteId site id
+ * @param {Number} postId post id
+ * @param {Number} commentId comment id
+ * @param {Author} author comment author
+ * @param {CommentInfo} info comment information
+ * @returns {Comment} represents a comment with its content
+ * @constructor
+ */
+export function LoadedComment( {
+	siteId = 0,
+	postId = 0,
+	commentId = 0,
+	author = null,
+	info = null,
+} ) {
+	const comment = new Comment( LoadedComment );
```

Comment:
Do we need another type for a pending/placeholder comment? Where a user has added a reply, but we aren't sure if it has been saved yet. Or do you think it would be okay to optimistically treat that as a LoadedComment (to the client).